packages/date-picker/src/panel/date.vue

```diff
@@ -318,7 +321,7 @@
 
       handleDatePick(value) {
         if (this.selectionMode === 'day') {
-          this.date = modifyDate(this.date, value.getFullYear(), value.getMonth(), value.getDate());
+          this.date = modifyWithDefaultTime(value, this.defaultTime);
```

Comment:
This resets time to `defaultTime` whenever a date is picked. Current behavior sets time only if time is not set.

Comparison: 
* (current) https://jsfiddle.net/zbgmnqod/1/
* (this pr) https://jsfiddle.net/bmxe9u8r/

@Leopoldthecoder What's your thoughts on this behavior?

Reply:
Might need https://github.com/ElemeFE/element/issues/9878 (breaking changes) to resolve behavior differences of panels elegantly. 😂🤔

Reply:
I think picking a date shouldn't change (reset) time.